stream.go

```diff
@@ -1465,6 +1480,9 @@ func (ss *serverStream) SetTrailer(md metadata.MD) {
 	if md.Len() == 0 {
 		return
 	}
+	if imetadata.Validate(md) != nil {
+		return
```

Comment:
@menghanl, should we do something else here besides silently omit the user's trailers?

What was the old behavior?

Reply:
This depends on the "invalid" characters I believe.
If it violates the wire protocol, it will result in a protocol error.
Otherwise, it will be sent.

But I'm not sure what characters are valid for http2, but invalid for gRPC headers.

Reply:
We should probably keep doing what we were doing before here, then, to avoid a behavior change?  I definitely don't think we should silently drop the trailers.

Reply:
> I definitely don't think we should silently drop the trailers.
I'm agree with @dfawley.May be we can printing a warning or error log, and adding a ` func (ss *serverStream) SetTrailer(md metadata.MD) error ` function on interface then deprecating the old function.

Reply:
> May be we can printing a warning or error log

An error log seems like the best option for now.  To add a new method would require a new name, could not be defined in the `ServerStream` interface, and we could never delete the old method due to our version policy.